cli/src/lib/semver.js

```diff
@@ -48,6 +48,75 @@ export function sortVersions(a: Version, b: Version): number {
   return 0;
 };
 
+/**
+ * Given a version range, returns the max version satisfying the range.
+ */
+function maxSat(ver: Version): ?Version {
+  if (ver.range === '>=') {
+    // TODO: ensure that if ver.upperBound exists, it is "greater than" ver
+    return ver.upperBound;
+  };
+  return ver;
+}
+
+/**
+ * Given a version range, returns the min version satisfying the range.
+ */
+function minSat(ver: Version): ?Version {
+  if (ver.range === '<=') {
+    // TODO: ensure that if ver.upperBound exists, it is "lesser than" ver
+    return ver.upperBound;
+  };
+  return ver;
+}
+
+/**
+ * Given two version ranges a and b, determine whether a is before b.
+ */
+function lt(n1: number | "x", n2: number | "x"): boolean | "maybe" {
+  if (typeof n1 === 'string' || typeof n2 === 'string') return false;
+  if (n1 < n2) return true;
+  if (n1 > n2) return false;
+  return "maybe";
+}
+function before(a: Version, b: Version): boolean {
+  let test = lt(a.major, b.major);
+  if (test !== "maybe") return test;
+  test = lt(a.minor, b.minor);
+  if (test !== "maybe") return test;
+  test = lt(a.patch, b.patch);
+  if (test !== "maybe") return test;
+  return false;
+}
+
+/**
+ * Given two versions, returns whether they are disjoint.
+ */
+function _before(a: ?Version, b: ?Version): boolean {
+  // If a is undefined, it denotes the maximum version. If b is undefined, it
+  // denotes the minimum version.
+  if (a && b) return before(a, b);
+  return false;
+}
+export function disjointVersions(a: Version, b: Version): boolean {
+  return _before (maxSat(a), minSat(b)) || _before (maxSat(b), minSat(a));
+}
+
+/**
+ * Given an array of versions, returns whether they are mutually disjoint.
+ */
+function _disjointVersionsAll(vers, i) {
+  const n = vers.length;
+  if (i+1 === n) return true;
+  for (var j = i+1; j < n; j++) {
+    if (!disjointVersions(vers[i], vers[j])) return false;
```

Comment:
No, I think it's correct. We check `vers[0]` with `vers[1], vers[2], ..., vers[n-1]`; then `vers[1]` with `vers[2], ..., vers[n-1]`, and so on. I'll add your test and rebase.